butterfree/pipelines/feature_set_pipeline_metadata.py

```diff
@@ -0,0 +1,220 @@
+    """
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    name: str = Field(..., description="The name of the column")
+    data_type: str = Field(
+        ...,
+        description=(
+            "The data type of the column (e.g., StringType, IntegerType) represented by pyspark.sql.types"  # noqa: E501
+        ),
+    )
+    description: str = Field(
+        ...,
+        description="The description of the column",
+    )
+    primary_key: bool = Field(
+        ...,
+        description="Whether the column is a primary (or partition if it's a Cassandra table) key",  # noqa: E501
+    )
+
+
+class Catalog(BaseModel):
+    """Metadata model for a feature set catalog.
+
+    This model represents the catalog information of a feature set,
+    including its name, description, and column definitions.
+    """
+
+    feature_set_name: str = Field(..., description="The name of the Feature Set")
+    description: str = Field(..., description="The description of the Feature Set")
+    columns: List[Column] = Field(..., description="A list of column definitions")
+
+
+class Metadata(BaseModel):
+    """Metadata model for a feature set pipeline.
+
+    This model represents the complete metadata of a feature set pipeline,
+    including its configuration, data sources, output schema, and processing details.
+    """
+
+    # To accept arbitrary types like FeatureSetPipeline
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    feature_set_pipeline: FeatureSetPipeline = Field(
+        ..., description="The feature set pipeline to create metadata from"
+    )
+    entity: str = Field(
+        ..., description="The entity type associated with the feature set"
+    )
+    is_incremental: bool = Field(
+        ..., description="The strategy for data generation (incremental or batch)"
+    )
+    feature_set_type: Literal["FeatureSet", "AggregatedFeatureSet"] = Field(
+        ..., description="The type of feature set"
+    )
+    windows_definition: Optional[List[str]] = Field(
+        None, description="The definition of the windows for the feature set"
+    )
+
+    # Required for correct serialization using Union
+    readers: List[
+        Annotated[
+            Union[FileReaderMetadata, KafkaReaderMetadata, TableReaderMetadata],
+            Field(discriminator="type"),
+        ]
+    ] = Field(
+        ...,
+        description="A list of data sources required to generate the feature set",
+    )
+
+    writers: List[WriterMetadata] = Field(
+        ..., description="The writers to be used for the feature set"
+    )
+    catalog: Catalog = Field(..., description="Metadata about the feature set's output")
+
+    @field_serializer("feature_set_pipeline")
+    def serialize_feature_set_pipeline(self, pipeline: FeatureSetPipeline) -> str:
+        """Serialize feature_set_pipeline to its class name."""
+        return pipeline.__class__.__name__
+
+    @classmethod
+    def _get_windows_definition(
```

Comment:
Does this make sense?

```suggestion
    def _get_time_window_definitions(
```

Reply:
I think it isn't necessary anymore